src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -358,7 +358,15 @@ static void GenerateXamlCodeBehind(XamlProjectItem? xamlItem, Compilation compil
 
 		sb.AppendLine("\t\tprivate void InitializeComponent()");
 		sb.AppendLine("\t\t{");
-		sb.AppendLine($"\t\t\tglobal::Microsoft.Maui.Controls.Xaml.Extensions.LoadFromXaml(this, typeof({rootType}));");
+		sb.AppendLine("\t\t\t[global::System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessage(\"TrimAnalysis\", \"IL2026:RequiresUnreferencedCode\",");
```

Comment:
If we know that this warning is only produced by the analyzer and should never be produced by the trimmer/AOT, then the correct way to suppress it is with `#pragma warning disable` - as that will act only inside the compiler and will not "pollute" the resulting IL with anything.
There are cases like this in other code bases (the analyzer doesn't understand something, but trimmer/AOT does) and we always tried to limit the suppression to pragmas only. The main goal is as usual code maintainability - if the code changes in the future and a real problem is introduced, trimmer would catch it. With the attribute in place, it would not.